### ISSUE TITLE
VM Config Disk resides in same datastore of template even when we provide storagepolicy in spec.

### ISSUE THREAD

**Issue by phogatsachin** [kubernetes-sigs/cluster-api-provider-vsphere#1858]
/kind bug

When vm are created by cluster api provider vsphere with datastore mentioned, hard disk is located in mentioned datastore but Configuration Disk for the vm is stored in same datastore as template used to clone the vm.

VSphereMachine Spec:
```yaml
spec:
  additionalDisksGiB: 100
  cloneMode: fullClone
  datacenter: DELDC1
  diskGiB: 100
  memoryMiB: 4096
  network:
    devices:
    - gateway4: 100.65.27.97
      ipAddrs:
      - 100.65.27.121/27
      nameservers:
      - 8.8.8.8
      networkName: P-95
  numCPUs: 2
  providerID: vsphere://421dab89-a309-ccf2-f8e0-ed5e2c9191ad
  storagePolicyName: NA-SSD-StoragePolicy
  template: rhel-7-kube-v1.22.15
```

![image](https://user-images.githubusercontent.com/129295571/229736680-4332027a-a444-4061-9882-aab7efe8dfa5.png)
As we can see the Configuration Disk resides in the datastore of template and Hard Disk are located in mentioned datastore. This is causing template datastore to fill up over period of time causing datastore being completely filled over period of time and all the vms running with that datastore to go into hung state.

**What did you expect to happen:**
All VM disk should be in the mentioned datastore.

- Cluster-api-provider-vsphere version: v1.5.1
- Kubernetes version: (use `kubectl version`): 1.22.15
- OS (e.g. from `/etc/os-release`): Rhel-7

**Comment by phogatsachin** [comment]
I have tested a fix locally. We have to make changes in pkg/services/govmomi/vcenter/clone.go file. Basically we need to set "spec.Location.Datastore = datastoreRef" not only for datastore use case, we should set this for all the cases, be it storagepolicy selected datastore or default datastore.

**Comment by chrischdi** [comment]
This should be fixed via #2134 .

/close

Feel free to reopen if there are still issues with this.
